pkg/helm/chartproxy/proxy.go

```diff
@@ -97,10 +112,18 @@ func (p *proxy) IndexFile(onlyCompatible bool) (*repo.IndexFile, error) {
 					}
 				}
 				if len(entry) > 0 {
+					if helmRepo.Overwrites != "" {
+						klog.Infof("%s overwrites repo %s", helmRepo.Name, helmRepo.Overwrites)
+						delKeys = append(delKeys, key+"--"+helmRepo.Overwrites)
```

Comment:
In the new `index.yaml` the keys are constructed with organization/vendor name as a prefix.  For example, it the org name is `acme` and the chart name is `awesome`, the key will be `acme-awesome`. Whereas the old repo uses just the name of the chart. So, here instead of using the key, you may need to use the chart name. Probably something like this: `entry[0].Name`.
```suggestion
						delKeys = append(delKeys, entry[0].Name+"--"+helmRepo.Overwrites)
```

Reply:
yeap, makes sense good catch